src/geogenalg/analyze.py

```diff
@@ -58,3 +64,119 @@ def calculate_coverage(
         columns=["base_feature_id", "overlay_area", "base_area"],
         errors="ignore",
     )
+
+
+def calculate_main_angle(polygon: Polygon) -> float:
+    """Calculate the main angle of a polygon based on its minimum bounding rectangle.
+
+    Args:
+    ----
+        polygon: A Shapely Polygon geometry
+
+    Returns:
+    -------
+        Orientation angle in degrees (range: 0 to 180), measured from the positive
+        Y-axis.
+
+    Raises:
+    ------
+        TypeError: If the input is not a Shapely Polygon.
+        ValueError: If the input polygon is empty.
+
+    """
+    if not isinstance(polygon, Polygon):
+        msg = "Input geometry must be a Shapely Polygon."
+        raise TypeError(msg)
+
+    if polygon.is_empty:
+        msg = "Input polygon is empty."
+        raise ValueError(msg)
+
+    min_rotated_rectangle = polygon.minimum_rotated_rectangle
```

Comment:
This does not need to be a variable, you can just directly do `list(polygon.minimum_rotated_rectangle.exterior.coords)`